src/app/projects/new-project/new-project.component.ts

```diff
@@ -0,0 +1,73 @@
+import { ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from "@angular/forms";
+import { ProjectsService } from "../projects.service";
+import { Project } from "../project";
+import { MatBottomSheetRef } from "@angular/material";
+import { HttpErrorResponse } from "@angular/common/http";
+import { GlobalErrorHandlerService } from "../../error/global-error-handler.service";
+
+@Component({
+  selector: 'app-new-project',
+  templateUrl: './new-project.component.html',
+  styleUrls: ['./new-project.component.scss']
+})
+export class NewProjectComponent implements OnInit, OnDestroy {
+
+  projectForm: FormGroup;
+  errorMessage: string = null;
+  addedProjectsCount = 0;
+
+  constructor(private projectService: ProjectsService,
+              private formBuilder: FormBuilder,
+              private bottomSheet: MatBottomSheetRef,
+              private changeDetector: ChangeDetectorRef,
+              private globalErrorHandlerService: GlobalErrorHandlerService) {
+    this.projectForm = formBuilder.group({
+      name: new FormControl(),
```

Comment:
It would be good to have `Validators.required` for the name field.

Reply:
done